app/controllers/registrations_controller.rb

```diff
@@ -17,4 +19,16 @@ def update_resource(resource, params)
       resource.update_with_password(params)
     end
   end
+
+  private
+
+  def register_mailing_list
+    if resource.persisted?
```

Comment:
We will only want to run this code with the api key is present, can we add a check for that please?

`if resource.persisted? && ENV['MAILCHIMP_API_KEY']`

Reply:
This would allow the application to create users that are not subscribed if the API key is not present. Is that ok? Should there be some kind of warning or logged notification that the environment variable is not set?

Reply:
Yeah it should be fine @BideoWego, this will allow us to skip this feature in development and sandbox environments. The environment variable should only be present in production.

Reply:
Ok, this sounds good. What about the unit test that will require the environment variable(s) to hit the API?

Reply:
I've updated my PR and it should address all the points you brought up. Let me know your thoughts. Thanks!